core/src/main/java/org/elasticsearch/index/query/FuzzyQueryBuilder.java

```diff
@@ -174,4 +217,72 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String getName() {
         return NAME;
     }
+
+    @Override
+    public Query doToQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        Query query = null;
+        MappedFieldType fieldType = parseContext.fieldMapper(fieldName);
+        if (fieldType != null) {
+            query = fieldType.fuzzyQuery(value, fuzziness, prefixLength, maxExpansions, transpositions);
+        }
+        if (query == null) {
+            int maxEdits = fuzziness.asDistance(BytesRefs.toString(value));
+            query = new FuzzyQuery(new Term(fieldName, BytesRefs.toBytesRef(value)), maxEdits, prefixLength, maxExpansions, transpositions);
```

Comment:
This comes from this PR https://github.com/elastic/elasticsearch/pull/12020/files#diff-8a2008f5cb78b0eecefab5ef414d7231R124. This is using the same logic as termQuery in the way it handles Object.